viewer/src/main/webapp/viewer-html/components/Snapping.js

```diff
@@ -0,0 +1,204 @@
+            id: this.name + 'Container',
+            width: '100%',
+            height: '100%',
+            autoScroll: true,
+            layout: {
+                align: 'stretch',
+                type: 'vbox'
+            },
+            style: {
+                backgroundColor: 'White'
+            },
+            renderTo: this.getContentDiv(),
+            items: [this.layerSelector]
+        });
+        this.inputContainer = Ext.getCmp(this.name + 'InputPanel');
+    },
+    /**
+     * handle checkbox events.
+     * @param {type} checkboxgroup
+     * @param {type} changedId
+     */
+    selectionChanged: function (checkboxgroup, changedId) {
+        var me = this;
+        if (this.snapCtl === null) {
+            this.createController();
+        }
+
+        if (!checkboxgroup.getValue().snaplayer) {
+            // nothing checked...
+            for (var i = 0; i < this.loadedLayerIds.length; i++) {
+                me.config.viewerController.setLayerVisible(
+                        me.config.viewerController.getAppLayerById(me.loadedLayerIds[i])
+                        , false);
+            }
+            me.snapCtl.removeAll();
+            me.snapCtl.deactivate();
+            me.loadedLayerIds = [];
+        }
+
+        checkboxgroup.items.each(function (item) {
+            // Retrieve appLayer from config.viewerController.
+            // Because the applayers in the comboBox are not the same as in the
+            // viewercontroller but copies. So by retrieving the ones from the
+            // ViewerController you get the correct appLayer
+            var appLayer = me.config.viewerController.getAppLayerById(item.inputValue);
+            var idx = me.loadedLayerIds.indexOf(item.inputValue);
+
+            if (item.checked) {
+                if (idx < 0) {
+                    console.debug("add data for layer", item.inputValue, appLayer.layerName);
+                    me.config.viewerController.setLayerVisible(appLayer, true);
+                    me.loadedLayerIds.push(item.inputValue);
+                    me.snapCtl.addLayerDataFor(appLayer);
+                    //} else {
+                    //    console.debug("nothing todo data already loaded for layer", item.inputValue, appLayer.layerName);
+                }
+            } else {
+                if (idx > -1) {
+                    console.debug("remove data for layer", item.inputValue, appLayer.layerName);
+                    me.config.viewerController.setLayerVisible(appLayer, false);
+                    me.loadedLayerIds.splice(idx, appLayer);
+                    me.snapCtl.removeLayer(appLayer);
+                    //} else {
+                    //    console.debug("nothing todo/remove for layer", item.inputValue, appLayer.layerName);
+                }
+            }
+        });
+    },
+    createController: function () {
+        this.snapCtl = Ext.create('viewer.components.OpenLayersSnappingController', {
```

Comment:
Zoals eerder gezegd is dit niet de plek waar dit geinstantieerd moet worden (in ieder geval niet de framework specifieke controller)